### DISCUSSION TITLE
BigQuery Insert

### DISCUSSION THREAD

**Discussion by jralduaveuthey** [hasura/graphql-engine#9043] [Question] [answered]
It looks like inserting data in a BigQuery table is not possible at the moment (see [here](https://hasura.io/docs/latest/mutations/index/) )
Is there any workaround to insert/modify data in a BigQuery table?

**Comment by adas98012** [comment, marked as answer]
You are right, Hasura doesn't support mutations on BigQuery database tables. Here is the info page on Hasura's current BigQuery support :
https://hasura.io/docs/latest/databases/bigquery/index/

You may look at other APIs to write into BigQuery database. Here is a source of reference:

https://cloud.google.com/bigquery/docs/write-api

**Reply by dsandip** [reply]
@jralduaveuthey Like @adas98012 said, you can look at other APIs and can also choose to expose any integrations or the APIs themselves as Actions. Would also love to have your feedback on the following:

1) Why write to bigquery via the GraphQL api? Would love to know the complete data lifecycle (generation, storage, consumption specifically).
2) What does your stack look like end to end?

**Comment by jralduaveuthey** [comment]
Thanks for the info!

Regarding our stack, we have an app developed in Adalo and our DB is now on
BQ. With Hasura we just would like to create personal Endpoints to be used
in Adalo


Viele Grüße / Best regards,
Jaime Raldua

**Reply by dameleney** [reply]
3 follow up questions:

> our DB is now on BQ

Is BigQuery the only DB you are using for the application? Are you using an OLTP DB in conjunction with BigQuery?
Given what you have described I assume your app is serving some sort of analytical solution that BigQuery is optimized for. However, I would assume you still need to access some OLTP information like user data.

>With Hasura we just would like to create personal Endpoints to be used in Adalo

When you say a personal endpoint I assume you are referring to a Hasura endpoint that uses Hasura's auth to return the data relevant to the user accessing the application. Are there any other details you can share?

How much data are you planning to read and write from BigQuery to enable your application?
